figures/scripts/supp-subtype-umap.R

```diff
@@ -0,0 +1,212 @@
+# for anything else, recode them as `Other CNS Tumor`
+umap_df_lgg <- umap_df %>%
+  dplyr::mutate(subtypes_to_plot = case_when(
+    grepl("LGG", molecular_subtype) ~ gsub("LGG, ", "", molecular_subtype),
+    grepl("GNT", molecular_subtype) ~ gsub("GNT, ", "", molecular_subtype),
+    grepl("GNG", molecular_subtype) ~ gsub("GNG, ", "", molecular_subtype),
+    TRUE ~ "Other CNS tumor"
+  )) %>% 
+  dplyr::mutate(cdkn_status = case_when(
+    grepl("CDKN2A/B", molecular_subtype) ~ "altered",
+    TRUE ~ "not altered"
+  )) %>% 
+  dplyr::mutate(subtypes_to_plot = case_when(
+    grepl("CDKN2A/B", subtypes_to_plot) ~ gsub(", CDKN2A/B", "", subtypes_to_plot),
+    TRUE ~ subtypes_to_plot
+  )) %>% 
+  dplyr::mutate(lgat_subtypes = case_when(
+    grepl("-somatic", subtypes_to_plot) & !grepl("-germline", subtypes_to_plot) ~ gsub("-somatic", "", subtypes_to_plot),
+    grepl("-germline", subtypes_to_plot) & !grepl("-somatic", subtypes_to_plot) ~ gsub("-germline", "", subtypes_to_plot),
+    grepl("-somatic, NF1-germline", subtypes_to_plot) ~ gsub("-somatic, NF1-germline", "", subtypes_to_plot),
+    TRUE ~ subtypes_to_plot
+  )) %>% 
+  group_by(lgat_subtypes) %>% 
+  dplyr::mutate(sample_n = n()) %>% 
+  dplyr::mutate(colored_subtype = case_when(
+    sample_n < 10 ~ "Other LGAT subtypes", 
+    TRUE ~ lgat_subtypes
+  ))
+
+# reorder the levels for plotting
+umap_df_lgg$cdkn_status <- factor(umap_df_lgg$cdkn_status, levels = c("not altered",
+                                                                      "altered"))
+
+umap_df_lgg$colored_subtype <- factor(umap_df_lgg$colored_subtype, levels = c("Other CNS tumor",
+                                                                              "To be classified",
+                                                                              "Other LGAT subtypes",
+                                                                              "BRAF V600E",
+                                                                              "KIAA1549-BRAF",
+                                                                              "NF1",
+                                                                              "other MAPK",
+                                                                              "RTK",
+                                                                              "wildtype"))
+
+# generate a plot
+p <- plot_dimension_reduction(umap_df_lgg,
+                              point_color = "colored_subtype",
+                              point_shape = "cdkn_status",
+                              x_label = "Dimension 1",
+                              y_label = "Dimension 2",
+                              alpha_value = 0.7,
+                              score1 = 2,
+                              score2 = 3,
+                              color_palette = c("#a9a9a9", "#656565", "#000000", palette_OkabeIto %>% c(1,2,3,5,7,8)))
```

Comment:
I see some hex colors being used a couple times - `"#a9a9a9"` and `"#656565"`, in particular. These should be defined as variables when `palette_OkabeIto` is made (around line 41) for use throughout the script, with meaningful names for what they refer to in the plots. Just in case something has to change later, this will make the code more flexible.

Reply:
@sjspielman thanks so much for the suggestion. They are left out of the palette_OkabeIto since those are the two different shades of grey that are used for either other CNS tumor or tumors to be classified. Hence I did not include them in the palette vector. 
However, to your point, I did add these as variables in the first part of the script so that the modification is easier.

Reply:
Yes, this is exactly what I had in mind!